src/main/java/de/qabel/desktop/Kernel.java

```diff
@@ -0,0 +1,232 @@
+        }
+    }
+
+    public void setDatabaseFile(Path databaseFile) {
+        this.databaseFile = databaseFile;
+    }
+
+    public void start() throws Exception {
+        PlatformUtils.start();
+        checkVersion();
+        new FXApplicationLauncher().launch(app);
+    }
+
+    public ResourceBundle getResources() {
+        return getContainer().getResourceBundle();
+    }
+
+    ButtonType cancelButton;
+    ButtonType updateButton;
+    AtomicReference<Alert> alertRef = new AtomicReference<>();
+    private void checkVersion() {
+        try {
+            if (currentVersion.equals("dev")) {
+                return;
+            }
+
+            if (!checker.isCurrent(currentVersion)) {
+                final boolean required = !checker.isAllowed(currentVersion);
+
+                String message = required ? getResources().getString("updateRequired") : getResources().getString("updatePossible");
+                PlatformImpl.runAndWait(() -> {
+                    Alert alert = new Alert(required ? WARNING : INFORMATION, message, cancelButton, updateButton);
+                    alertRef.set(alert);
+                    alert.setHeaderText(null);
+                    alert.showAndWait().ifPresent(buttonType -> {
+                        if (buttonType == updateButton) {
+                            try {
+                                documentLauncher.accept(checker.getDesktopVersion().getDownloadURL());
+                                shutdown.run();
+                            } catch (Exception e) {
+                                logger.error("failed to open download: " + e.getMessage(), e);
+                            }
+                        }
+                        if (required) {
+                            shutdown.run();
+                        }
+                    });
+                });
+            }
+        } catch (Exception e) {
+            logger.error("failed to check for updates: " + e.getMessage(), e);
+        }
+    }
+
+    private void shutdown() {
+        PlatformUtils.shutdown();
+        System.exit(-1);
+    }
+
+    private ClientDatabase getConfigDatabase() {
+        try {
+            try {
+                Path configDir = databaseFile.toAbsolutePath().getParent();
+                if (!Files.isDirectory(configDir)) {
+                    Files.createDirectory(configDir);
+                }
+                connection = DriverManager.getConnection(getSqliteConnectionString());
+
+                try {
+                    try (Statement statement = connection.createStatement()) {
+                        statement.execute("PRAGMA FOREIGN_KEYS = ON");
+                    }
+                    ClientDatabase clientDatabase = new DesktopClientDatabase(connection);
+                    clientDatabase.migrate();
+
+                    return clientDatabase;
+                } catch (Exception e) {
+                    try { connection.close(); } catch (Exception ignored) {}
```

Comment:
nope. the ClientDatabase is used with that connection. Only if the setup / migration failes, the connection is useless. otherwise it is used in the whole application with that connection.